src/environment/tests.go

```diff
@@ -0,0 +1,279 @@
+package environment
+
+import (
+	"fmt"
+	"github.com/gojinja/gojinja/src/runtime"
+	"github.com/gojinja/gojinja/src/utils/slices"
+	"reflect"
+	"strings"
+)
+
+func getInt(idx int, values ...any) (int64, error) {
+	if len(values) <= idx {
+		return 0, fmt.Errorf("not enough values passed to the test")
+	}
+	res, ok := toInt(values[idx])
+	if !ok {
+		return 0, fmt.Errorf("value passed to the test is not an integer")
+	}
+	return res, nil
+}
+
+func toInt(v any) (i int64, ok bool) {
+	if p, ok := v.(int); ok {
+		return int64(p), ok
+	}
+	if p, ok := v.(int8); ok {
+		return int64(p), ok
+	}
+	if p, ok := v.(int16); ok {
+		return int64(p), ok
+	}
+	if p, ok := v.(int32); ok {
+		return int64(p), ok
+	}
+	if p, ok := v.(uint); ok {
+		return int64(p), ok
+	}
+	if p, ok := v.(uint8); ok {
+		return int64(p), ok
+	}
+	if p, ok := v.(uint16); ok {
+		return int64(p), ok
+	}
+	if p, ok := v.(uint32); ok {
+		return int64(p), ok
+	}
+	if p, ok := v.(uint64); ok {
+		return int64(p), ok
```

Comment:
uint64 can be out of scope for int64; shouldn't we check for this case?

Reply:
We have to draw a line somewhere on size of the ints, this way it's easier and honestly who would use 64 uint in a template.  In this case including uint64 would not complicate the code but in operator the code would get much more complex I don't think it's worth it at this stage of the project.

Reply:
> who would use 64 uint in a template

we have to remember that python supports bignums, so somebody for sure used that

> In this case including uint64 would not complicate the code too much

I just thought about doing

```
if uintValue64 out of int64 scope {
  return 0, error("value out of scope for int64")
}
``` 
and cascade the error down 

----

> but in operator the code would get much more complex 

ok i get the point, let's leave it as it is